Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -942,4 +1162,22 @@ private URI getDummyImagePath() throws URISyntaxException, IOException {
             throw new IOException("No dummy image found in resources!");
         }
     }
+
+    /**
+     * Concatenate two given Strings representing a directory and a filename and return the result. A file separator
+     * is added between both Strings if 'directory' does not already end with a separator character.
+     *
+     * @param directory
+     *          directory path
+     * @param filename
+     *          filename
+     * @return directory concatenated with filename
+     */
+    public static String concat(String directory, String filename) {
+        if (directory.endsWith("/")) {
```

Comment:
I guess `if (directory.endsWith(File.separator))` would be a little more elegant and should work too, shouldn't it?

Reply:
I guess, it really depends how this String is created. If it comes from some malformed URI.create or some other not really defined source it can work incorrectly. 

Maybe instead of concat we could use here some Paths.get(directory, filename).toString()?

Reply:
You can use both in Java (`C:\Program Files (x86)\Java` = `C:/Program Files (x86)/Java`), so we don’t know what is configured.

Reply:
> Maybe instead of concat we could use here some Paths.get(directory, filename).toString()?

This sounds like the most sensible solution. I changed the code accordingly and tested it.